* Drop gdm_clearenv_no_lang() which clears the environment but keeps the i18n
  environment variables.
* Switch a "buf" declaration from char to unsigned char.

XXX To be splitted further and clarified

--- gdm-2.16.4.orig/daemon/misc.c
+++ gdm-2.16.4/daemon/misc.c
@@ -344,37 +344,6 @@
 }
 
 /*
- * Clear environment, but keep the i18n ones,
- * note that this leaks memory so only use before exec
- * (keep LD_* if preserve_ld_vars is true)
- */
-void
-gdm_clearenv_no_lang (void)
-{
-	int i;
-	GList *li, *envs = NULL;
-
-	for (i = 0; environ[i] != NULL; i++) {
-		char *env = environ[i];
-		if (strncmp (env, "LC_", 3) == 0 ||
-		    strncmp (env, "LANG", 4) == 0 ||
-		    strncmp (env, "LINGUAS", 7) == 0)
-			envs = g_list_prepend (envs, g_strdup (env));
-		if (preserve_ld_vars &&
-		    strncmp (env, "LD_", 3) == 0)
-			envs = g_list_prepend (envs, g_strdup (env));
-	}
-
-	ve_clearenv ();
-
-	for (li = envs; li != NULL; li = li->next) {
-		putenv (li->data);
-	}
-
-	g_list_free (envs);
-}
-
-/*
  * Clear environment completely
  * note that this leaks memory so only use before exec
  * (keep LD_* if preserve_ld_vars is true)
@@ -1391,7 +1360,7 @@
 int
 gdm_fdgetc (int fd)
 {
-	char buf[1];
+	unsigned char buf[1];
 	int bytes;
 
 	VE_IGNORE_EINTR (bytes = read (fd, buf, 1));
